### PR TITLE
fix: release: Enrolling E2EI crash [WPB-6788]

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6788" title="WPB-6788" target="_blank">BugWPB-6788</a>  [Android] Fix crash on opening the browser for E2EI IDP authorization
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

Crash while enrolling E2EI certificate. 

### Causes (Optional)

Enrolling the certificate `GetE2EICertificateUseCase` requires the `context` which is Activity for `startActivityForRresult`, but with recent changes context that it gets is not an Activity but an Application. 

### Solutions

Remove `GetE2EICertificateUseCase` and move all the logic of enrolling the E2EI certificate into a separate Compose to be able to provide Activity from it and re-use in all the places.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3265](https://10.10.124.134/job/AR-build-pipeline/3265/) **failed**.

**Comment by AndroidBob** [comment]
Build [3271](https://10.10.124.134/job/AR-build-pipeline/3271/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8965392)(8965392)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8965392)(8965392)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [3337](https://10.10.124.134/job/AR-build-pipeline/3337/) **failed**.

**Review by MohamadJaara — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/e2eiEnrollment/GetE2EICertificateUI.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    val coroutineScope = rememberCoroutineScope()
+    val context = LocalContext.current
+
+    // FIXME issue happens when this UI is called from WireActivity: WebView is just canceled by itself
+    LaunchedEffect(Unit) {
+        viewModel.requestOAuthFlow.onEach {
+            OAuthUseCase(context, it.target, it.oAuthClaims, it.oAuthState).launch(
+                context.getActivity()!!.activityResultRegistry
+            ) { result -> viewModel.handleOAuthResult(result, it) }
+        }.launchIn(coroutineScope)
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/e2eiEnrollment/GetE2EICertificateUI.kt:48]: question for next PR:
why we don't use
`val launcher = rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
if (result.resultCode == Activity.RESULT_OK) {
//  you will get result here in result.data
}
}`

**Review by mchenani — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3346](https://10.10.124.134/job/AR-build-pipeline/3346/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.1(9006504)(9006504)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.1(9006504)(9006504)-staging-compat.apk)

**Pull request merged**